### PR TITLE
Add AMQP Adapter

### PR DESCRIPTION
**Pull request by tinque**
I added a AMQP Adpater. (tested with RabbitMQ)

### CODE REVIEW

**Comment by tinque** [comment]
Problem with params uri (not enough test) sorry

**Comment by ekryski** [comment]
Oh hell yeah! Stoked to check this out!

**Comment by daffl** [comment]
This is great! Do you have an idea how we could test this on Travis?

**Comment by tinque** [comment]
To be honest ! I have no idea ...

**Comment by ekryski** [comment]
I think I do. I've done a bunch with RabbitMQ.

File: src/amqp.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    listSubscriber[ev] = [cb];
+  }
+};
+
+var publish = function(ev,data){
+  var msg = {};
+  msg.ev = ev;
+  msg.data = data;
+  //amqp name
+  if(ch){
```
Comment (ekryski) [inline src/amqp.js]: I'd like to see these variables longer. I know that `ch` means `channel` but people new to RabbitMQ may not.

File: src/sync.js [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 var redis = require('./redis');
+var amqp = require('./amqp');
 
 module.exports = function(config) {
 
+  var proto = '';
   config = config || { db: 'mongodb://localhost:27017/sync' };
 
-  var proto = config.db.split('://')[0];
+  if(config.db){
```
Comment (ekryski) [inline src/sync.js]: Maybe we should just change this to `config.uri` since it's no longer just dbs anymore. @daffl?

**Comment by ekryski** [comment]
@tinque this is solid! Thanks! I made a couple minor comments but I am also fine making those changes and looking into getting tests running on travis. This is good to 🚢 !  🎉  🍻

File: src/amqp.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          debug('Error while assertQueue to AMQP: ', err.toString());
+        }else{
+          ch.bindQueue(q.queue, exchangeName, '');
+           if(typeof cb === 'function'){
+             cb(ch);
+           }
+          ch.consume(q.queue, function(msg) {
+            var temp = JSON.parse(msg.content.toString());
+            onmessage(temp.ev,temp.data);
+          }, {noAck: true});
```
Comment (ekryski) [inline src/amqp.js]: We may later on, want to turn on "acks" but for now this is totally fine.

**Comment by ekryski** [comment]
I'll figure out what's going on with the tests. I'll probably add some more and fix them up over the next couple days.

**Comment by petermikitsh** [comment]
@ekryski @daffl Can we get a release of feathers-sync that includes this functionality? I'm stoked to get my feathers app to fly. Thanks!

**Pull request merged**